src/raw/mod.rs

```diff
@@ -29,6 +72,100 @@ impl<K, V> Table<K, V> {
         self.bins.len()
     }
 
+    pub(crate) fn get_moved<'g>(
+        &'g self,
+        for_table: Shared<'g, Table<K, V>>,
+        guard: &'g Guard,
+    ) -> Shared<'g, BinEntry<K, V>> {
+        match self.next_table(guard) {
+            t if t.is_null() => {
+                // if a no next table is yet associated with this table,
+                // create one and store it in `self.next_table`
+                match self.next_table.compare_and_set(
+                    Shared::null(),
+                    for_table,
+                    Ordering::SeqCst,
+                    guard,
+                ) {
+                    Ok(_) => {}
+                    Err(changed) => {
+                        assert!(!changed.current.is_null());
+                        assert_eq!(changed.current, for_table);
+                    }
+                }
+            }
+            t => {
+                assert_eq!(t, for_table);
+            }
+        }
+        // return a shared pointer to BinEntry::Moved
+        self.moved.load(Ordering::SeqCst, guard)
+    }
+
+    pub(crate) fn find<'g, Q>(
+        &'g self,
+        bin: &BinEntry<K, V>,
```

Comment:
Come to think of it, now that this method is on `Table`, can't we just pass the `bini` here instead? I guess taking the `&BinEntry` directly saves us a load potentially?

Reply:
We could, but both in `get_node` and in `find` (the places that call `find`) there is a `null` check before. In `find`, we also `deref()` and `match` the `bin` to decide whether we found a `Node` or need to follow a `Moved` to a next table. 

This implementation is has a double `match` anyways due to the recursive call to `find`, but if we change this to `bini` we would `load` and `deref` the bin anyways (for the `match`), only to then pass `bini` and `load` and `deref` again. 

I don't see a huge advantage in this. If anything, one would want to `find` _only with_ hash and key. But `hash()` is on the map and doing this is basically what `get_node` is for, though I agree that it would be the nicer interface for a method called `find`. But maybe this is more due name and less due to function?